content/en-us/scripting/security/security-tactics.md

```diff
@@ -144,6 +197,17 @@ Imagine a game where a player can fire a laser beam at another player. Rather th
 - Confirm that the hit player is alive.
 - Store weapon and player state on the server and confirm that a firing player is not blocked by a current action such as reloading or a state like sprinting.
 
+#### DataStore Manipulation
+
+In games using `Class.DataStoreService` to save player data, exploiters may take advantage of invalid [data](#data-validation), and more obscure methods, to prevent a DataStore from saving properly. This can be especially abused in games with item trading, marketplaces and similar systems where items or currency leave a player's inventory.
+
+Ensure that any actions performed through a `Class.RemoteEvent` or `Class.RemoteFunction` that affect player data with client input is sanitised based on the following:
+
+- `Class.Instance` values cannot be serialised into a DataStore and will fail. Utilise [type validation](#remote-runtime-type-validation) to prevent this.
+- DataStores have a data size limit per key of 4 megabytes. Strings of arbitrary length should be checked and/or capped to avoid this, alongside ensuring limitless arbitrary keys cannot be added to tables by the client.
+- Table indices cannot be [NaN](#data-validation) or nil. Iterate over all tables passed by the client and verify all indices are valid.
+- DataStores can only accept valid UTF-8 characters. Sanitise all strings provided by the client to ensure only bytes 1-127 are used. By using `Library.string.find()` with the pattern `"[%z\128-\255]"` an operation can be aborted if any invalid characters are found in a string.
```

Comment:
```suggestion
- Table indices cannot be `NaN` or `nil`. Iterate over all tables passed by the client and verify all indices are valid.
- `Class.DataStore|DataStores` can only accept valid UTF-8 characters. Sanitize all strings provided by the client to ensure only bytes 1-127 are used. By using `Library.string.find()` with the pattern `"[%z\128-\255]"`, an operation can be aborted if any invalid characters are found in a string.
```

Reply:
I linked back to data-validation here as it provides functions to check for NaN. If someone jumps to this point of the article there's a chance they could miss this function. That's pretty unlikely though, so this change is fine.